atlp-rwanda/script-heroes-bn-backend
# #174797518 Login using email and password

#### What does this PR do?
Enable signin using email and password
#### Description of Task to be completed?
- Checking for email password matching
- Providing the user with JW token 
#### How should this be manually tested?
- Clone the repo `git clone https://github.com/atlp-rwanda/script-heroes-bn-backend.git`
- Switch your working `cd script-heroes-bn-backend`
- Checkout the branch `git checkout ft-email-login-#174797518`
- Create a `.env` file in the root folder and update it using data found in `env-example`
- Install all dependencies `npm install`
- Run migrations `sequelize db:migrate`
- Start the server `npm run dev`
- Using postman on Signup endpoint create an account on the app
    - Endpoint for signup ```http://localhost:3000/api/auth/signup``` 
- Still on postman Login using Email and password used to create the account
    - Endpoint for login ```http://localhost:3000/api/auth/login```
> You should see successful login message and the returned token
#### Any background context you want to provide?
- Number 3000 in endpoints may depend on the port you put in your `.env` 
- Data for creating an account should look like:
```json
{
    "firstName": "firstName",
    "lastName": "lastName",
    "email": "email@test.com",
    "phoneNumber": "0780000000",
    "password": "password"
}
```
- Data for login should look like
```json
{
    "email": "email@test.com",
    "password": "password"
}
```
- Response after login should look like
```json
{
    "message": "Your are successfully loged in",
    "token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MTksImZpcnN0TmFtZSI6IkhhYmluZXphIiwibGFzdE5hbWUiOiJqYW52aWVyIiwiZW1haWwiOiJoYWJpbmV6YWphbnZAZ21haWwuY29tIiwicGhvbmVOdW1iZXIiOiIwNzg5MTIxMzI0IiwiaWF0IjoxNjAxMzk5NTM5LCJleHAiOjE2MDE0ODU5Mzl9.sGPkCF17q8sl6AibrrFcdSYfdsjoVkJuMIxPeO5X0dE"
}
```
#### What are the relevant pivotal tracker stories?
[#174797518](https://www.pivotaltracker.com/story/show/174797518)
#### Screenshots if availabe
![image](https://user-images.githubusercontent.com/44426710/94598968-0f4a3600-0290-11eb-9eb8-4a36b0a4ad04.png)
